index.bs

```diff
@@ -137,55 +241,45 @@ available [=command|commands=] by module and name.
-   entries are the details of the event that has occurred.
+ - A <dfn export for=event>event type</dfn> which is defined by a [=local
+   end schema=] fragment containing a group. Each such group has two fields:
+    - <code>name</code> which is a string literal of the form <code>[module
+      name].[event name]</code>. This is the <dfn export for=event>event
+      name</dfn>.
 
-Event [=event/parameter type|parameter types=] are defined in this
-specification using [[!RFC8610|CDDL]].
+    - <code>params</code> which defines a mapping containing event data. The
+      populated value of this map is the <dfn export for=command>event
+      parameters</code>.
 
 ## Processing Model ## {#processing-model}
 
-<div algorithm>
-To <dfn>process a command</dfn> given a qualified command name |qualified command name|,
-and parameters |parameters|:
-
-   1. If |qualified command name| does not contain exactly one U+002E
-      FULL STOP character (.), return an [=Error=] with [=error code=]
-      [=unknown command=].
-
-   1. Let |module name| be the bytes in |qualified command name|
-      preceeding the U+002E FULL STOP character.
-
-   1. Let |command name| be the bytes in |qualified command name|
-      following the U+002E FULL STOP character.
 
-   1. Let |command| be the command listed in the [=table of commands=]
-      whose module is |module name| and whose name is |command name|.
-
-   1. If there is no such |command|, return an [=Error=] with [=error
-      code=] [=unknown command=].
+<div algorithm>
+To <dfn>process a command</dfn> given |data|:
 
-   1. If |parameters| does not [=match a CDDL specification|match the
-      CDDL specification=] given by |command|'s [=command/parameter
-      type=], then return an [=Error=] with [=error code=] [=invalid
-      argument=].
+   1. Match |data| against the [=remote end schema=]. If this results in a
+      match, return the result of [=trying=] to run the [=remote end steps=]
+      for the command with the [=command name=] equal to the <code>name</code>
+      property of |data| and with the [=command parameters=] from the matched
+      data.
 
-   1. Let |result| be the result of [=trying=] to run the [=remote end
-      steps=] for |command| with |parameters|.
+   1. Otherwise there is no match. If |data| isn't a map or is a map without a
+      property named </code>method</code> return an [=Error=] with error code
+      [=invalid argument=]. Otherwise let <var>command</var> be the value of
+      the <code>method</code> property of |data|.
 
-   1. [=Assert=]: |result| [=match a CDDL specification|matches the CDDL
-      specification=] given by |command|'s [=command/return type=].
+   1. If |command| is not in the [=set of all command names=], return an
+      [=Error=] with [=error code=] [=unknown command=].
 
-   1. Return [=success=] with data |result|.
+   1. Return an [=Error=] with [=error code=] [=invalid argument=].
```

Comment:
Is this last step correct? (It could be a default "return an error", assuming there are other situations for returning an invalid args error different from step 2).

Also in step 2, how happens with `command`? Should that get returned? Or something else?

Reply:
I don't think I understand the question. The basic structure of this algorithm (with slightly different numbering) is:

1. Try to match the input against the schema.
2. If the match succeeds, then run the remote end steps for the command that matched and return that result.
3. [In all further steps it must be true that the match failed]
4. Check if we got something that looks like a map containing a method name at all and if not return invalid argument error
5. If we have a method name and it's not a known method return invalid command error
6. Otherwise we have a valid method name but the schema didn't match so return an invalid argument error.

Reply:
> Otherwise we have a valid method name but the schema didn't match so return an invalid argument error.

Got it, I think step 6 is what I was trying to describe as "default return an error", so forgive the noise.